### PR TITLE
Add extra settings for deleting cookie

### PR DESCRIPTION
**Pull request by nmenezes0**
This PR amends the logout mechanism to ensure that the domain is specified when deleting the cookie

This has been tested in dev with this PR https://github.com/i-dot-ai/ask-ai/tree/test/delete-cookies, note that:
1. in this instance the `COLA_COOKIE_DOMAIN` does not begin with a '.'
2. there are no other uses of the `COLA_COOKIE_DOMAIN`

### CODE REVIEW

File: automatilib/cola/views.py [hunk 1]
```diff
@@ -142,11 +142,15 @@ def get(
...
                 permanent_redirect = redirect(reverse(settings.COLA_LOGIN_FAILURE), status=401)
-                permanent_redirect.delete_cookie(settings.COLA_COOKIE_NAME)
+                response.delete_cookie(
+                    settings.COLA_COOKIE_NAME, domain=".cabinetoffice.gov.uk", secure=True, httponly=True
+                )
                 return permanent_redirect
 
             temporary_redirect = HttpResponse("Unauthorized", status=401)
-            temporary_redirect.delete_cookie(settings.COLA_COOKIE_NAME)
+            response.delete_cookie(
```
Comment (gecBurton) [inline automatilib/cola/views.py:151]: this line should be removed (its the wrong response)

Comment (gecBurton) [inline automatilib/cola/views.py]:
```suggestion
            temporary_redirect.delete_cookie(
```

File: automatilib/cola/views.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ def get(self, request: HttpRequest, **kwargs: dict) -> HttpResponse:
 
         redirect_url = settings.LOGIN_URL
         response = redirect(reverse(redirect_url))
-        response.delete_cookie(settings.COLA_COOKIE_NAME)
+        response.delete_cookie(settings.COLA_COOKIE_NAME, domain=".cabinetoffice.gov.uk")
```
Comment (252afh) [inline automatilib/cola/views.py]: We have the domain in a variable, `COLA_COOKIE_DOMAIN`

File: automatilib/cola/views.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def get(self, request: HttpRequest, **kwargs: dict) -> HttpResponse:
...
         return response
```
Comment (252afh) [inline automatilib/cola/views.py]: Could we move the `logout` to after the cookie has been deleted. If something goes wrong they could be logged out with a valid cookie still

**Review by 252afh — CHANGES_REQUESTED** [review]

File: automatilib/cola/views.py [hunk 4]
```diff
@@ -142,11 +142,11 @@ def get(
             LOGGER.error(f"cookie error: {error}")
             if getattr(settings, "COLA_LOGIN_FAILURE", None) is not None:
                 permanent_redirect = redirect(reverse(settings.COLA_LOGIN_FAILURE), status=401)
-                permanent_redirect.delete_cookie(settings.COLA_COOKIE_NAME)
+                permanent_redirect.delete_cookie(settings.COLA_COOKIE_NAME, domain=".cabinetoffice.gov.uk")
```
Comment (gecBurton) [inline automatilib/cola/views.py:145]:
```suggestion
                permanent_redirect.delete_cookie(settings.COLA_COOKIE_NAME, domain=".cabinetoffice.gov.uk", secure=True, httponly=True)
```

File: automatilib/cola/views.py [hunk 5]
```diff
@@ -142,11 +142,11 @@ def get(
...
                 return permanent_redirect
 
             temporary_redirect = HttpResponse("Unauthorized", status=401)
-            temporary_redirect.delete_cookie(settings.COLA_COOKIE_NAME)
+            temporary_redirect.delete_cookie(settings.COLA_COOKIE_NAME, domain=".cabinetoffice.gov.uk")
```
Comment (gecBurton) [inline automatilib/cola/views.py:149]:
```suggestion
            temporary_redirect.delete_cookie(settings.COLA_COOKIE_NAME, domain=".cabinetoffice.gov.uk", secure=True, httponly=True)
```

File: automatilib/cola/views.py (hunk 2 above)
Comment (gecBurton) [inline automatilib/cola/views.py:67]:
```suggestion
        response.delete_cookie(settings.COLA_COOKIE_NAME, domain=".cabinetoffice.gov.uk", secure=True, httponly=True)
```

File: automatilib/cola/views.py (hunk 3 above)
Comment (gecBurton) [reply automatilib/cola/views.py]: good idea [9cd314c](https://github.com/i-dot-ai/automatilib/pull/27/commits/9cd314c5fe8948e2ea5fc874e9f12d2f21989cbd)

**Review by 252afh — APPROVED** [review]

**Pull request merged**